src/main/java/stsjorbsmod/powers/FragilePower.java

```diff
@@ -16,23 +19,41 @@ public FragilePower(final AbstractCreature owner, final int turnsUntilSnap) {
 
         this.owner = owner;
         this.amount = turnsUntilSnap;
+        this.amount2 = MemoryManager.forPlayer(owner).countCurrentClarities();
+        this.type = SPECIAL;
+
+        updateDescription();
+    }
+
+    @Override
+    public void onGainClarity(String memoryID) {
+        this.amount2 = MemoryManager.forPlayer(owner).countCurrentClarities();
+        updateDescription();
+    }
 
+    @Override
+    public void onLoseClarity(String memoryID) {
+        this.amount2 = MemoryManager.forPlayer(owner).countCurrentClarities();
         updateDescription();
     }
 
+    @Override
+    public void onSnap() {
+        addToBot(new RemoveSpecificPowerAction(owner, owner, ID));
+    }
+
     @Override
     public void atEndOfTurn(boolean isPlayer) {
         if (this.amount == 1) {
-            AbstractDungeon.actionManager.addToBottom(new SnapAction(owner, true));
-            AbstractDungeon.actionManager.addToBottom(new RemoveSpecificPowerAction(this.owner, this.owner, this.ID));
+            addToBot(new SnapAction(owner, true));
         } else {
-            AbstractDungeon.actionManager.addToBottom(new ReducePowerAction(this.owner, this.owner, this.ID, 1));
+            addToBot(new ReducePowerAction(this.owner, this.owner, this.ID, 1));
         }
     }
 
     @Override
     public void updateDescription() {
-        description = String.format(this.amount == 1 ? DESCRIPTIONS[0] : DESCRIPTIONS[1], this.amount);
+        description = String.format(this.amount == 1 ? DESCRIPTIONS[0] : DESCRIPTIONS[1], this.amount, amount2 * 3, amount2 * 6);
```

Comment:
Better to make public and reuse the constants in SnapAction than repeating the 3/6 magic numbers here